cachi2/core/package_managers/gomod.py

```diff
@@ -0,0 +1,1022 @@
+    :return: a list containing all non-existing go.mod files across subpaths
+    :rtype: list
+    """
+    invalid_gomod_files = []
+    for subpath in subpaths:
+        package_gomod_rel_path = source_path / subpath / "go.mod"
+        log.debug("Testing for go mod file in {}".format(package_gomod_rel_path))
+        if not package_gomod_rel_path.exists():
+            invalid_gomod_files.append(package_gomod_rel_path)
+
+    return invalid_gomod_files
+
+
+def _resolve_gomod(path: Path, request: Request, git_dir_path=None):
+    """
+    Resolve and fetch gomod dependencies for given app source archive.
+
+    :param str path: the full path to the application source code
+    :param dict request: the Cachito request this is for
+    :param list dep_replacements: dependency replacements with the keys "name" and "version"; this
+        results in a series of `go mod edit -replace` commands
+    :param RequestBundleDir git_dir_path: the full path to the application's git repository
+    :return: a dict containing the Go module itself ("module" key), the list of dictionaries
+        representing the dependencies ("module_deps" key), the top package level dependency
+        ("pkg" key), and a list of dictionaries representing the package level dependencies
+        ("pkg_deps" key)
+    :rtype: dict
+    :raises GoModError: if fetching dependencies fails
+    """
+    if git_dir_path is None:
+        git_dir_path = request.source_dir
+
+    worker_config = get_worker_config()
+
+    with GoCacheTemporaryDirectory(prefix="cachito-") as temp_dir:
+        env = {
+            "GOPATH": temp_dir,
+            "GO111MODULE": "on",
+            "GOCACHE": temp_dir,
+            "PATH": os.environ.get("PATH", ""),
+            "GOMODCACHE": "{}/pkg/mod".format(temp_dir),
+        }
+
+        if worker_config.cachito_athens_url:
+            athens_url = worker_config.cachito_athens_url
+            env["GOPROXY"] = f"{athens_url}|{athens_url}"
+
+        if "cgo-disable" in request.flags:
+            env["CGO_ENABLED"] = "0"
+
+        run_params = {"env": env, "cwd": path}
+
+        # Collect all the dependency names that are being replaced to later verify if they were
+        # all used
+        replaced_dep_names = set()
+        for dep_replacement in request.dep_replacements:
+            name = dep_replacement["name"]
+            replaced_dep_names.add(name)
+            new_name = dep_replacement.get("new_name", name)
+            version = dep_replacement["version"]
+            log.info("Applying the gomod replacement %s => %s@%s", name, new_name, version)
+            _run_gomod_cmd(
+                ("go", "mod", "edit", "-replace", f"{name}={new_name}@{version}"), run_params
+            )
+        # Vendor dependencies if the gomod-vendor flag is set
+        flags = request.flags
+        should_vendor, can_make_changes = _should_vendor_deps(
+            flags, request.source_dir, worker_config.cachito_gomod_strict_vendor
```

Comment:
I think some of these paths are wrong. Everything that was `app_subpath` before should be `path`, not `request.source_dir`

request.source_dir is the root of the repo, `path` can be a subpath within that repo
```suggestion
            flags, path, worker_config.cachito_gomod_strict_vendor
```

Reply:
Same for the one passed to `_module_lines_from_modules_txt()`, and I think those are the only two occurrences

Reply:
Fixed both ocurrences.